src/include/partition_filter.h

```diff
@@ -43,17 +43,15 @@ typedef struct
 } ResultRelInfoHolder;
 
 
-/* Forward declaration (for on_new_rri_holder()) */
+/* Forward declaration (for on_rri_holder()) */
 struct ResultPartsStorage;
 typedef struct ResultPartsStorage ResultPartsStorage;
 
 /*
- * Callback to be fired at rri_holder creation.
+ * Callback to be fired at rri_holder creation/destruction.
```

Comment:
IMHO there should be 2 callbacks: one for creation, and one for destruction. Why would we want to mix these roles?